enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,272 @@
+As the last step, she removes the `service.beta.kubernetes.io/load-balancer-source-ranges` 
+service annotation.
+
+### Goals
+
+Enable the configuration of a range for a load balancer via the `IngressController.Spec`, 
+specifically via the `LoadBalancerStrategy`, with a new parameter `AllowedSourceRanges`, 
+which represents a list of IP address ranges. This is then used by the operator to set
+`.Spec.LoadBalancerSourceRanges` of the LoadBalancer Service in front of the router 
+in cloud environments.
+
+Enable the migration of users who set the `service.beta.kubernetes.io/load-balancer-source-ranges`
+annotation to the new IngressController API field.
+
+### Non-Goals
+
+This enhancement does not change the behavior or API for LoadBalancer-type services 
+in general (that is, services that aren't managed by the ingress operator).
+
+## Proposal
+
+The IngressController API is extended by adding an optional field `AllowedSourceRanges` 
+of type `[]CIDR` to `LoadBalancerStrategy`. If this field is empty, all source addresses 
+are allowed (which is equivalent to specifying "0.0.0.0/0" for IPv4 and "::/0" for IPv6).
+
+[openshift/api#1222](https://github.com/openshift/api/pull/1222) contains the literal API 
+changes that we're proposing for this EP.
+
+### Workflow Description
+
+1. A cluster admin who wants to limit access to the load balancer to a list of IP address ranges
+of their choice sets the `AllowedSourceRanges` field. 
+2. The ingress operator reconciles the LoadBalancer Service and sets the `.Spec.LoadBalancerSourceRanges`
+based on `AllowedSourceRanges`.
+
+#### Variation [optional]
+
+If a cluster admin already set `service.beta.kubernetes.io/load-balancer-source-ranges` annotation,
+the workflow differs to be a migration workflow.
+
+1. A cluster admin ensures that `service.beta.kubernetes.io/load-balancer-source-ranges` is set and 
+`.Spec.LoadBalancerSourceRanges` is unset.
+2. She upgrades her cluster to OpenShift v4.12.
+3. She sets `AllowedSourceRanges` on the IngressController. 
+4. The ingress operator then sets `Spec.LoadBalancerSourceRanges` based on `AllowedSourceRanges`.
+5. She removes `service.beta.kubernetes.io/load-balancer-source-ranges` service annotation.
+
+### API Extensions
+
+This proposal will modify the `IngressController` API by adding a new
+field called `AllowedSourceRanges` to the `LoadBalancerStrategy` 
+struct type and a new type `CIDR` that is a string in CIDR notation.
+
+[openshift/api#1222](https://github.com/openshift/api/pull/1222) contains the literal API changes that we're
+proposing for this EP.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The value set for `AllowedSourceRanges` will be used to set `.Spec.LoadBalancerSourceRanges` 
+of the LoadBalancer Service in front of the router in cloud environments.
+
+If `.Spec.LoadBalancerSourceRanges` is set and `AllowedSourceRanges` is empty, 
```

Comment:
or did I get service field and the ingress field backwards?  if you could clarify in all usages, it would really help less familiar readers.  I didn't even realize they were separate until we spoke in person.